src/renderer/MainContent.tsx

```diff
@@ -1,32 +1,42 @@
 import { Fragment, useMemo } from 'react';
 import { MAX_NUM_FILTERS, MIN_NUM_FILTERS } from 'common/constants';
+import { computeAvgFreq } from 'common/utils';
 import FrequencyBand from './components/FrequencyBand';
 import { useAquaContext } from './utils/AquaContext';
 import './styles/MainContent.scss';
 import AddSliderDivider from './components/AddSliderDivider';
 import Spinner from './icons/Spinner';
 
 const MainContent = () => {
-  const { filters: freqSortedFilters, isLoading } = useAquaContext();
+  const { filters, isLoading } = useAquaContext();
 
   // Store widths for AddSliderDividers and FrequencyBands so we can manually position them
   const DIVIDER_WIDTH = 28;
   const BAND_WIDTH = 72.94;
 
-  const [idSortedFilters, sortIndexMap] = useMemo(() => {
+  const [idSortedFilters, freqSortedFilters, sortIndexMap] = useMemo(() => {
     // Obtain a fixed order list of the filters
-    const fixedSort = freqSortedFilters
-      .slice()
-      .sort((a, b) => a.id.localeCompare(b.id));
+    const fixedSort = Object.values(filters).sort((a, b) =>
+      a.id.localeCompare(b.id)
+    );
+
+    // Obtain a visually sorted list of the filters
+    const visualSort = Object.values(filters).sort(
+      (a, b) =>
+        a.frequency - b.frequency ||
+        a.gain - b.gain ||
+        a.quality - b.quality ||
+        a.type.localeCompare(b.type)
```

Comment:
You could consider moving the arrow function into the `sortHelper` util function (and essentially replace the existing functionality since we aren't really using it anymore). Or you could delete the `sortHelper` function (assuming we aren't using it elsewhere).